flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/mapper/FlatModelMapper.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.mapper;
+
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.types.Row;
+
+import java.util.List;
+
+/**
+ * Abstract class for flatMappers with model.
+ * FlatModelMapper transform one Row type data into zero, one, or more Row type result data.
+ * Operations that produce multiple strictly one Row type result data per Row type data
+ * can also use the {@link ModelMapper}.
+ */
+public abstract class FlatModelMapper extends FlatMapper {
+
+	/**
+	 * schema of the model with Table type.
+	 */
+	protected TableSchema modelSchema;
+
+	public FlatModelMapper(TableSchema modelSchema, TableSchema dataSchema, Params params) {
+		super(dataSchema, params);
+		this.modelSchema = modelSchema;
+	}
+
+	/**
+	 * Load model from the list of Row type data.
+	 *
+	 * @param modelRows the list of Row type data
+	 */
+	public abstract void loadModel(List <Row> modelRows);
+
+	/**
+	 * Generate new instance of given FlatModelMapper class without model data.
+	 * The instance can not deal with real data, but it could be used to get the output result schema.
+	 *
+	 * @param flatModelMapperClassName Name of the FlatModelMapper class
+	 * @param modelScheme              The schema of input Table type model.
```

Comment:
it is not immediately clear to me what's the difference between `modelScheme` &  `dataSchema`. 
Also why is modelScheme of the type `TableSchema` ?

Reply:
Thanks, the model is the machine learning model that use the Table as its representation (serialized to Table from the memory or deserialized from Table to memory). Thus, the model need `modelSchema` and the predict data need `dataSchema`.

Reply:
ah. I see now. I think the prhase "the schema of the input Table" part confused me.
I would probably rephrase it to
```
@param modelScheme    The model scheme represented in Table format
```